server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutes.java

```diff
@@ -173,14 +192,78 @@ public void define(Service service) {
         @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500, message = "Internal server error - Something went bad on the server side.")
     })
     private TaskIdDto userActions(Request request, Response response) throws JsonExtractException {
-        UserVaultAction requestedAction = extractUserVaultAction(request);
+        VaultAction requestedAction = extractVaultAction(request);
+        validateAction(requestedAction, ImmutableList.of(VaultAction.RESTORE, VaultAction.EXPORT));
 
-        Task requestedTask = generateTask(requestedAction, request);
+        Task requestedTask = generateUserTask(requestedAction, request);
         TaskId taskId = taskManager.submit(requestedTask);
         return TaskIdDto.respond(response, taskId);
     }
 
-    private Task generateTask(UserVaultAction requestedAction, Request request) throws JsonExtractException {
+    @POST
+    @Path(ROOT_PATH)
+    @ApiOperation(value = "Purge all expired messages based on retentionPeriod of deletedMessageVault configuration")
+    @ApiImplicitParams({
+        @ApiImplicitParam(
+            required = true,
+            name = "action",
+            dataType = "String",
+            paramType = "query",
+            example = "?action=purge",
+            value = "Compulsory. Needs to be a purge action")
+    })
+    @ApiResponses(value = {
+        @ApiResponse(code = HttpStatus.CREATED_201, message = "Task is created", response = TaskIdDto.class),
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Bad request - action is invalid"),
+        @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500, message = "Internal server error - Something went bad on the server side.")
+    })
+    private TaskIdDto globalActions(Request request, Response response) {
+        Task vaultTask = generateVaultTask(request);
+        TaskId taskId = taskManager.submit(vaultTask);
+        return TaskIdDto.respond(response, taskId);
+    }
+
+    @DELETE
```

Comment:
Should be POST... 

 IMO `DELETE` returns 201

Here we create a related REST resource (a task) so a POST seems appropriate. The fact that this tasks actually DELETE the original endoint resources is somehow a side effect that might (hopefully) arise in the future resulting from this task execution.

IMO let's look at some work of people better than us at REST https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-delete-by-query.html : DeleteByQuery is a POST in ElasticSearch.

Agree with @rouazana on this: **it should be a POST**